web/src/main/java/com/google/sps/servlets/SaveStockServlet.java

```diff
@@ -45,22 +49,57 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
     KeyFactory keyFactory = datastore.newKeyFactory().setKind("Stock");
+    ArrayList<String> tikkers = new ArrayList<String>();
+    for (int i = 0; i < tik.size(); i++) {
+      String tikk = tik.get(i).text();
+      tikkers.add(tikk);
+    }
+
 
     for (int i = 0; i < tik.size(); i++) {
+      Key myTiks = datastore.newKeyFactory().setKind("Stock").newKey(tikkers.get(i));
 
-      String tikk = tik.get(i).text();
       String pricee = price.get(i).text().replaceAll("[\\\\$,]", "");
-      double priceDouble = Double.parseDouble(pricee);
+      Double priceDouble = Double.parseDouble(pricee);
 
-      FullEntity taskEntity =
-          Entity.newBuilder(keyFactory.newKey())
-              .set("Tik", tikk)
-              .set("TimeStamp", timeStamp)
+      Entity taskEntity =
+          Entity.newBuilder(datastore.get(myTiks))
+              .set("Tik", tikkers.get(i))
               .set("USD", priceDouble)
+              .set("TimeStamp", timeStamp)
+              .build();
+      datastore.update(taskEntity);
+    }
+
+    for (int i = 0; i < tik.size(); i++) {
+
+      String pricee = price.get(i).text().replaceAll("[\\\\$,]", "");
+      Double priceDouble = Double.parseDouble(pricee);
+
+      Query<Entity> query =
+          Query.newEntityQueryBuilder()
+              .setKind("Stock")
+              .setFilter(PropertyFilter.eq("Tik", tikkers.get(i)))
               .build();
-      datastore.put(taskEntity);
+      QueryResults<Entity> results = datastore.run(query);
+
+      Entity entity = results.next();
+
+      String tiko = entity.getString("Tik");
+      Long id = entity.getKey().getId();
+      Key keyy = entity.getKey();
```

Comment:
maybe a more descriptive variable name than "keyy". What key is this? What is it used for.